### ISSUE TITLE
Some windows system files are hidden from BleachBit

### ISSUE THREAD

**Issue by livingentity** [bleachbit/bleachbit#1240]
**Describe the bug**
BleachBit is unable to even see some system files in the first place which makes cleaning them impossible.

**To Reproduce**
1. Make a cleaner for all files in a directory like e.g. "%WinDir%\System32\config" or "%WinDir%\System32\sru".
    Alternatively use: winapp2.ini -> Microsoft Windows -> Windows Subsystems
2. Click on Preview for your cleaner. See that most or all of the files in the folders given above are unlisted. (Folders are listed if you use "walk.all", but their content is not.)

The error happens every time.

**Expected behavior**
BleachBit should list all files in the given directory.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - BleachBit version 4.4.0 Portable, 4.4.1 (Build 1976) Portable

**Comment by rados** [comment]
Seem related with #1234.

**Comment by livingentity** [comment]
@rados
Yes it's related. Can you reproduce this or is it an isolated issue on my system?

**Comment by rados** [comment]
@livingentity 
Yes, I've reproduced it but couldn't figure out the reason at the first glance. It needs examination.

**Comment by livingentity** [comment]
I found this: Running whoami /priv in windows command line shows different privileges than running it from inside BleachBit using the process action, even though _both are elevated_. Does that help or is it unrelated to this issue?

**Comment by livingentity** [comment]
Here's the solution. Sorry for having forgotten about this and opening an unnecessary issue:

BleachBit is 32-bit but I'm running Windows 64-bit.
In this situation Windows automatically redirects everything from %WinDir%\System32 to %WinDir%\SysWOW64.
However if you actually need to access %WinDir%\System32 you can easily do so by using _%WinDir%\Sysnative_ instead.

**Comment by livingentity** [comment]
I reopen this because after reconsidering I think it's a major issue.

Even though I can now solve this for my own cleaners, BleachBit doesn't apply the correction itself when using winapp2.ini (current version has 88 occurences of system32!), the included cleaners or the internal cleaners written in python.
In short BleachBit only cleans 32-bit files on Windows 64-bit if the path includes %windir%\system32.

This should also be clearly stated in the description of BleachBit:
It does not clean any 64-bit files in %windir%\system32 by default (though it technically could).

Why isn't there a compatibility correction in place like for ProgramFiles and CommonProgramFiles?
For winapp2.ini if the user runs Windows 64-bit replace all %windir%\system32 by %windir%\sysnative.
Also add a default variable $$System32$$ that expands to both %windir%\system32 and %windir%\sysnative (and change all cleaners to use it, of course).

Or am I misunderstanding something?

**Comment by az0** [comment]
This is good information, and I targeted it to 4.6.0 for a closer look. A potential solution is dynamic translation, especially for Winapp2.ini.

**Comment by odibowdi** [comment]
Hi,

Each time I launch BleachBit portable, there is a creation (copy) of a
[share.zip](https://github.com/bleachbit/bleachbit/files/7008959/share.zip)
[share.zip](https://github.com/bleachbit/bleachbit/files/7008962/share.zip)
 folder on the Desktop, containing the fonts from the folder “share”, after checking the folder is still in the “share”folder???
I deleted the folder on the Desktop (more than 5x), and it is created as soon as I launch BleachBit (still in the share folder of the app).
I'm using it with Windows 10 21H1 Build 19043.1165.

**Comment by rados** [comment]
@odibowdi 
Thanks for this report. You can create a new issue specifically for this, so it would be tracked in an easier way.

**Comment by rados** [comment]
@odibowdi
One quick thing that you can try is to completely uninstall the Bleachbit application if you have installed it. Then reset the machine and run the portable version.

**Comment by odibowdi** [comment]
Hi Rados,

Thank You for your answer,

I didn't use the installer but the portable one, after deleting it completely, I rebooted, then downloaded a new one (portable), and after unzipping it... the issue was gone (copy of existing share folders on the Desktop), but there's a new one, e.g.: I cannot scroll down the cleaning options (the scrolling bar isn't there!), unless I select one item and “Previsualize” it first, once I clicked on “Previsualize”, it brings me back to the top of the list in the left pane, then the scrolling bar appears and works, then next, I have to scroll down to the previous selection, and after checking if it's the one selected, then I can clean it.
Like the previous issue, it happens on two different laptops, a Toshiba Satellite i7-5500U, 16 GO of ram, SSD Sandisk Ultra 3D 1TO, and the other one is a DELL XPS-15 7590, i9-9980HK, 64 GO of ram, and a SSD SK Hynix NVMe PC601, 1TO.

So, the first issue happened on both laptops, and after deleting the portable version (with a search in the “Registry's Hives”, for some remaining traces, on both laptops), rebooted, downloaded & installed the portable version, rebooted again, and execute it as “Admin”, and both laptops get the same issue !???
Both laptops are using Windows 10 Professional 21H1 Build 19043.1165

![Capture d’écran 2021-08-22 093904](https://user-images.githubusercontent.com/39953370/130355900-855103d0-f0cb-4451-bfc5-62d64d46fe23.png)
![Animation](https://user-images.githubusercontent.com/39953370/130355904-6296b282-6620-4f8e-ab9d-27fb584b13e7.gif)
![Animation (1)](https://user-images.githubusercontent.com/39953370/130355907-d5931cbd-782c-4e84-99db-7c52f66b216a.gif)

Thank You

**Comment by rados** [comment]
@odibowdi 
This is a part of the known issues that has already been resolved but it is not part of an official release yet. You can check this description:

https://www.bleachbit.org/news/bleachbit-440#known_issues
